Blockzilla/AppDelegate.swift

```diff
@@ -104,6 +106,65 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    func application(_ application: UIApplication, open url: URL, sourceApplication: String?, annotation: Any) -> Bool {
+        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
+            return false
+        }
+
+        guard let urlTypes = Bundle.main.object(forInfoDictionaryKey: "CFBundleURLTypes") as? [AnyObject],
+            let urlSchemes = urlTypes.first?["CFBundleURLSchemes"] as? [String] else {
+                // Something very strange has happened; org.mozilla.Blockzilla should be the zeroeth URL type.
+                return false
+        }
+
+        guard let scheme = components.scheme, let host = url.host, urlSchemes.contains(scheme) else {
+            return false
+        }
+
+        let query = getQuery(url: url)
+
+        switch host {
+        case "open-url":
+            let url = unescape(string: query["url"]) ?? ""
+            if application.applicationState == .active {
+                // If we are active then we can ask the BVC to open the new tab right away.
+                // Otherwise, we remember the URL and we open it in applicationDidBecomeActive.
+                browserViewController.submit(url: URL(string: url)!)
```

Comment:
let's guard the force unwrapping just to be safe `guard let url = url else { return false }` etc. etc.